lms/static/js/commerce/views/receipt_view.js

```diff
@@ -164,7 +164,7 @@ var edx = edx || {};
          * @return {object} JQuery Promise.
          */
         getCourseData: function (courseId) {
-            var courseDetailUrl = '/api/course_structure/v0/courses/%s/';
+            var courseDetailUrl = '/api/courses/v1/courses/%s/';
```

Comment:
That's strange. Maybe it's because we don't include a cookie in the request made by the JS? You need to be [authenticated](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/commerce/views.py#L34) to see the receipt page, so I'd have expected the username to be required as per the API's documentation. (I don't like that the username is required, but that's a separate issue altogether.)